plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,512 @@
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: interface_group_policy_1
+    state: query
+  register: query_with_name
+
+- name: Query all existing L3Out interface group policies under l3out
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    state: query
+
+- name: Delete an existing L3Out interface group policy with name
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: interface_group_policy_1
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data, check_if_all_elements_are_none
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True, aliases=["l3out_template"]),
+        l3out=dict(type="str", required=True, aliases=["l3out_name"]),
+        description=dict(type="str"),
+        name=dict(type="str", aliases=["l3out_interface_group_policy"]),
+        interface_routing_policy=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+                template=dict(type="str"),
+            ),
+            required_together=[
+                ["name", "template"],
+            ],
+        ),
+        interface_routing_policy_uuid=dict(type="str"),
+        bfd=dict(
+            type="dict",
+            options=dict(
+                enable=dict(type="bool"),
+                authentication=dict(type="bool"),
+                key_id=dict(type="int"),
+                key=dict(type="str", no_log=True),
+            ),
+        ),
+        bfd_multi_hop=dict(
+            type="dict",
+            options=dict(
+                enable=dict(type="bool"),
+                authentication=dict(type="bool"),
+                key_id=dict(type="int"),
+                key=dict(type="str", no_log=True),
+            ),
+        ),
+        ospf=dict(
+            type="dict",
+            options=dict(
+                enable=dict(type="bool"),
+                authentication_type=dict(type="str", choices=["simple", "md5", "none"]),
+                key_id=dict(type="int"),
+                key=dict(type="str", no_log=True),
+            ),
+        ),
+        qos_priority=dict(type="str", choices=["level1", "level2", "level3", "level4", "level5", "level6", "unspecified"]),
```

Comment:
if the values of QOS_LEVEL in constants.py is same as qos_priority, can we re-use this?